src/common/core/core.js

```diff
@@ -1659,6 +1663,173 @@ define([
+         *
+         * @param {module:Core~Node} node - the node in question.
+         *
+         * @return {bool} - Returns true if your node is a library element, false otherwise.
+         *
+         * @func
+         */
+        this.isLibraryElement = core.isLibraryElement;
+
+        /**
+         * Returns the fully qualified name of any node, which is the list of its namespaces separated
+         * by comma and followed by the name of the node.
+         *
+         * @param {module:Core~Node} node - the node in question.
+         *
+         * @return {string} - Returns the fully qualified name of the node (like: 'library.myName').
+         *
+         * @func
+         */
+        this.getFullyQualifiedName = core.getFullyQualifiedName;
+
+        /**
+         * Removes a library from your project. It will also remove any remaining instances of the specific library.
+         *
+         * @param {module:Core~Node} node - any node in your project.
+         * @param {string} name - the name of your library.
+         *
+         * @func
+         */
+        this.removeLibrary = core.removeLibrary;
+
+        /**
+         * Returns the origin GUID of any library node.
+         *
+         * @param {module:Core~Node} node - the node in question.
+         *
+         * @return {GUID | null} - Returns the origin GUID of the node or null if the node is not a library element.
+         *
+         * @func
+         */
+        this.getLibraryGuid = core.getLibraryGuid;
+
+        /**
+         * Rename a library in your project.
+         *
+         * @param {module:Core~Node} node - any node in your project.
+         * @param {string} oldName - the current name of the library.
+         * @param {string} newName - the new name of the project.
+         *
+         * @func
+         */
+        this.renameLibrary = core.renameLibrary;
+
+        /**
+         * Returns the info associated with the library.
+         *
+         * @param {module:Core~Node} node - any node in the project.
+         * @param {string} name - the name of the library.
+         *
+         * @return {object} - Returns the information object, stored alongside the library (that basically
+         * carries metaData about the library).
+         *
+         * @func
+         */
+        this.getLibraryInfo = core.getLibraryInfo;
+
+        /**
+         * Returns the core node of the given library.
+         *
+         * @param {module:Core~Node} node - any node in the project.
+         * @param {string} name - the name of the library.
+         *
+         * @return {module:Core~Node | null} - Returns the library root node or null, if invalid library was queried.
+         *
+         * @func
+         */
+        this.getLibraryRoot = core.getLibraryRoot;
+
+        /**
+         * Returns all the Meta nodes found in the library.
+         *
+         * @param {module:Core~Node} node - any node of your project.
```

Comment:
yes. I know that is could seem odd, but as these informaitons are always cached, I believe it is more useful if you can query with any arbitrary node (you just need the name of the library)